File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/GuildHouseUpdateInformationMessage.scala





















// Generated on 05/03/2013 10:34:50
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildHouseUpdateInformationMessage extends MessageDeserializer {
  val messageId = 6181

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GuildHouseUpdateInformationMessage.messageId

    val housesInformations: HouseInformationsForGuild.TypeImpl

    def serialize(buf: Buffer) {
      housesInformations.serialize(buf)
    }

  }

  protected def create(housesInformations: HouseInformationsForGuild.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val housesInformations = HouseInformationsForGuild.deserialize(buf)

    create(housesInformations)
  }

}

object GuildHouseUpdateInformationMessage extends GuildHouseUpdateInformationMessage {
  case class MessageImpl(housesInformations: HouseInformationsForGuild.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(housesInformations: HouseInformationsForGuild.TypeImpl) = MessageImpl(housesInformations)

  def apply(housesInformations: HouseInformationsForGuild.TypeImpl) = create(housesInformations)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildHouseUpdateInformationMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildHouseUpdateInformationMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6181;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public HouseInformationsForGuild housesInformations;
    
    public GuildHouseUpdateInformationMessage() { }
    
    public GuildHouseUpdateInformationMessage(HouseInformationsForGuild housesInformations) {
        this.housesInformations = housesInformations;
    }
    
    @Override
    public void serialize(Buffer buf) {
        housesInformations.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        housesInformations = new HouseInformationsForGuild();
        housesInformations.deserialize(buf);
    }
    
}
